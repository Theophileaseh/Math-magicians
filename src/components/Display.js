import React from 'react';
import PropTypes from 'prop-types';

import '../styles/Display.css';

function Display({ total, next, operation }) {
  return (
    <div className="main-screen">
      <span className="screen">{total}</span>
      <span className="screen">{operation}</span>
      <span className="screen">{next}</span>
    </div>
  );
}

Display.propTypes = {
  total: PropTypes.string,
  next: PropTypes.string.isRequired,
  operation: PropTypes.string.isRequired,
};

Display.defaultProps = {
  total: '0',
};

export default Display;
