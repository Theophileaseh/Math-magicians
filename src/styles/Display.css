.main-screen {
  height: 75px;
  background: #ffffff;
  width: 400px;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  margin: 0 auto;
  border-radius: 8px;
}

.screen {
  color: #000;
  font-size: 25px;
  margin: 15px;
}
