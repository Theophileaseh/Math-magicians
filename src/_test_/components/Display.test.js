import React from 'react';
import renderer from 'react-test-renderer';

import Display from '../../components/Display';

it('matches display snapshot', () => {
  const tree = renderer.create(
    <Display />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
