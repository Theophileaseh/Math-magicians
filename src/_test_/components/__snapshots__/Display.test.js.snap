// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches display snapshot 1`] = `
<div
  className="main-screen"
>
  <span
    className="screen"
  >
    0
  </span>
  <span
    className="screen"
  />
  <span
    className="screen"
  />
</div>
`;
